templates/types/streaming/nextjs/app/components/ui/chat/hooks/use-file.ts

```diff
@@ -112,34 +112,19 @@ export function useFile() {
 
     const filetype = docMineTypeMap[file.type];
     if (!filetype) throw new Error("Unsupported document type.");
-    const newDoc: Omit<DocumentFile, "content"> = {
-      id: uuidv4(),
-      filetype,
+    const uploadedFileMeta = await uploadContent(file, requestParams);
+    const newDoc: DocumentFile = {
+      id: uploadedFileMeta.id,
```

Comment:
it's weird that one object has two times the same ID (`id` and `metadata.id`)

Reply:
yes, i kept the `id` to avoid conflict with old code in TS. But now it's already removed.